src/AppProvider.js

```diff
@@ -0,0 +1,91 @@
+import React, { useState } from "react";
+import { v4 as uuid } from "uuid";
+import { useInterval, usePersistedState } from "./utils/hooks";
+import { translateToSeconds } from "./utils/helpers";
+
+export const AppContext = React.createContext({});
+
+export const AppProvider = ({ children }) => {
+
+    const [timers, setTimers] = usePersistedState('timers', []);
+    const [paused, setPaused] = useState(true);
+    const [activeIndex, setActiveIndex] = useState(0);
+    const [currentTime, setCurrentTime] = useState(0);
+    const [isComplete, setIsComplete] = useState(false);
+    const [currentRound, setCurrentRound] = useState(1);
+    const [onlyEnableStart, setOnlyEnableStart] = useState(true);
+
+    useInterval(() => {
```

Comment:
this centralization of logic is great - all the logic lives here so the rest becomes very simple